apps/user-ms/src/main/java/it/pagopa/selfcare/user/controller/UserController.java

```diff
@@ -249,16 +249,44 @@ public Uni<Void> updateUserProductStatus(@PathParam("id") String userId,
                 .onItem().transformToUni(loggedUser -> userService.updateUserProductStatus(userId, institutionId, productId, status, loggedUser));
     }
 
-    @Operation(summary = "Create or Update user")
+
+
+    /**
+     * The createOrUpdateByUserId function is used to update existing user adding userRole.
+     *
+     * @param userId Sting
+     * @param userDto CreateUserDto
+     */
+    @Operation(summary = "Create user by userId - Using for addUserRole")
     @ResponseStatus(HttpStatus.SC_NO_CONTENT)
     @POST
+    @Path("/{userId}")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
-    public Uni<Response> createOrUpdate(@Valid CreateUserDto userDto) {
-        return userService.createOrUpdateUser(userDto)
+    public Uni<Response> createOrUpdateByUserId(@PathParam("userId") String userId,
+                                                @Valid AddUserRoleDto userDto) {
+        return userService.createOrUpdateUserByUserId(userDto, userId)
                 .map(ignore -> Response.status(HttpStatus.SC_NO_CONTENT).build());
     }
 
+    /**
+     * The createOrUpdateByFiscalCode function is used to create a new user or update an existing one.
+     * The function takes in a CreateUserDto object, which contains the following fields:
+     * - fiscalCode (String): The tax code of the user. This field is required and must be unique for each user.
+     * - email (String): The email address of the user. This field is optional and can be null or empty string if not provided by caller.
+     * - phoneNumber (String): The phone number of the user, including country code prefix (+39). This field is optional and can be null or empty string if not provided by
+     *
+     * @param userDto CreateUserDto
```

Comment:
good.....you could add this in "summary" so it will be also showed in open api spec